src/Bicep.Core/Analyzers/Linter/Rules/UseProtectedSettingsForCommandToExecuteSecretsRule.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.Analyzers.Linter.Common;
+using Bicep.Core.Parsing;
+using Bicep.Core.Resources;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Metadata;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.Visitors;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class UseProtectedSettingsForCommandToExecuteSecretsRule : LinterRuleBase
+    {
+        public new const string Code = "use-protectedsettings-for-commandtoexecute-secrets";
+
+        public UseProtectedSettingsForCommandToExecuteSecretsRule() : base(
+            code: Code,
+            description: CoreResources.UseProtectedSettingsForCommandToExecuteSecretsRuleDescription,
+            docUri: new Uri($"https://aka.ms/bicep/linter/{Code}")
+        )
+        { }
+
+        public override string FormatMessage(params object[] values)
+            => string.Format(CoreResources.UseProtectedSettingsForCommandToExecuteSecretsRuleMessage, (string)values[0]);
+
+        public override IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel semanticModel)
+        {
+            List<IDiagnostic> diagnostics = new List<IDiagnostic>();
+
+            foreach (ResourceMetadata resource in semanticModel.AllResources)
+            {
+                ResourceDeclarationSyntax resourceSyntax = resource.Symbol.DeclaringResource;
+                if (resourceSyntax.TryGetBody()?.SafeGetPropertyByName("properties") is ObjectPropertySyntax propertiesSyntax
+                    && propertiesSyntax.Value is ObjectSyntax properties)
+                {
+                    if (properties.SafeGetPropertyByName("type")?.Value is StringSyntax typeSyntax
+                        && typeSyntax.TryGetLiteralValue() is string typePropertyValue)
+                    {
+                        if (typePropertyValue.Contains("CustomScript", StringComparison.OrdinalIgnoreCase))
```

Comment:
Here's how to find out the possible values:
```sh
 az vm extension image list
```
Gives a list like:
```json
  {
    "name": "DSC",
    "publisher": "Microsoft.Powershell.Test01",
    "version": "1.0.0.0"
  },
  {
    "name": "CustomScriptForLinux",
    "publisher": "Microsoft.OSTCExtensions",
    "version": "1.0"
  },
...
```

I've made changes to make the code more specific to publisher/type and less specific to resource type (because there's no definite way to determine that).

I could theoretically add a unit test to call "az vm extension image list" and verify PublisherAndNameList, but that would require the Azure CLI be installed, so I'm hesitant.  Thoughts?

Reply:
@majastrz ?

Reply:
No, let's not add a CLI dependency just yet. Can always revisit if it becomes a problem.